classes/class-p4-master-site.php

```diff
@@ -1103,4 +1106,34 @@ public function add_help_sidebar() {
 			$screen->set_help_sidebar( $sidebar );
 		}
 	}
+
+	/**
+	 * Generate structured data for a search engine.
+	 */
+	public function generate_p4_structured_data() {
+		$p4_site_url   = get_site_url();
+		$p4_site_title = get_bloginfo( 'name' );
+
+		// Filter social menu links.
+		$footer_social_menu = wp_get_nav_menu_items( 'Footer Social' );
+		$social_menu_links  = [];
+
+		foreach ( $footer_social_menu as $menu_details ) {
+			$social_menu_links[] = $menu_details->url;
+		}
+
+		$p4_structured_data  = '<script type=\'application/ld+json\'>';
```

Comment:
@sagarsdeshmukh the string should not be constructed here. Please use a twig template, pass the needed variables to context and then use them to include or not the new template for the structured data.